play-json-ops-common-213/src/main/scala/play/api/libs/json/ops/v4/Exceptions.scala

```diff
@@ -0,0 +1,128 @@
+package play.api.libs.json.ops.v4
+
+import play.api.libs.json._
+
+import scala.reflect._
+import scala.reflect.runtime.universe._
+
+private[ops] object Exceptions {
+
+  /**
+    * Returns the estimated literal string of code you would use to match on this value.
+    */
+  def keyAsCodeLiteral(keyValue: Any): String = {
+    keyValue match {
+      case s: String => "\"%s\"".format(s) // for some reason string interpolation goes wonky here in Scala 2.11
+      case _ => keyValue.toString // this should work for most case classes and primitive types
+    }
+  }
+
+  /**
+    * Returns the path to the key in a more code-friendly format.
+    */
+  def keyPathAsString(keyPath: JsPath): String = {
+    val str = keyPath.toJsonString
+    if (str == "obj") "__" else str
+  }
+}
+
+/**
+  * An exception representing the failure to match a type key to the specific serializer
+  * for that type.
+  *
+  * @param valueType the type object of the generic type that could not be serialized
+  * @param keyValue the value of the key that was unmatched with a serializer
+  * @param keyPath the path to the key
+  * @param keyType the type object of the key
+  */
+class UnrecognizedTypeKeyException(
+  val valueType: Type,
+  val keyValue: Any,
+  val keyPath: JsPath,
+  val keyType: Type
+  ) extends RuntimeException({
+  val keyValueStr = Exceptions.keyAsCodeLiteral(keyValue)
+  val keyPathString = Exceptions.keyPathAsString(keyPath)
+  val keyPathWithType = s"$keyPathString.as[$keyType]"
+  s"Cannot parse instance of $valueType.\n" +
+    s"Unrecognized key '$keyValue' as extracted at path with type $keyPathWithType\n" +
```

Comment:
HINT: Some log ingestion tools can automatically parse data when it's formatted like `key=$keyValue` or `key='$keyValue'`. I find it also helps force you to give a good "label" to every piece of data you include in the message, and usually any log parsing logic will be safe if/when you tweak the wording of the message (not to mention if you ever localize).

Reply:
I'm a little worried about changing the behavior here from the other cross-compiled versions. I think we should address this in a separate PR if it is an issue.